kedro/framework/session/session.py

```diff
@@ -379,7 +398,8 @@ def run(  # pylint: disable=too-many-arguments,too-many-locals
         )
 
         try:
-            run_result = runner.run(filtered_pipeline, catalog, hook_manager, run_id)
+            run_result = runner.run(filtered_pipeline, catalog, hook_manager)
+            self._run_called = True
```

Comment:
So this is just in case the run is successful. If there's a problem with the pipeline halfway through, we can call `session.run()` again. Is that the intended course? Makes sense for debugging, if a little confusing for the user (but I've run it multiple times to failure, it allowed me to that then!).

Reply:
Yes, so my thinking was that if you're in an interactive workflow to experiment/debug your pipeline it would be overhead if you need to recreate a new `session` if the pipeline didn't run successfully. At the same time this is a niche case just for those people that would actually do a `run` inside a notebook/ipython session.